### PR TITLE
Sup 6216 add checkout ssh secret

### PR DESCRIPTION
**Pull request by lizrabuya**
- Add `ssh_secret`  property under the `checkout` definition
- Add tests to validate name and reject an empty `ssh_secret` name
- Add examples in `test/valid-pipelines/checkout.yml`

### CODE REVIEW

File: schema.json [hunk 1]
```diff
@@ -227,6 +227,12 @@
             { "fetch": "--prune --tags" },
             { "clone": "", "fetch": "" }
           ]
+        },
+        "ssh_secret": {
+          "type": "string",
+          "description": "Name of SSH private key secret from Buildkite Secrets to use for git clone/fetch operations. The secret must contain a valid SSH private key. The key will be written to a temporary file with 0400 permissions and removed after checkout completes.",
```
Comment (petetomasik) [inline schema.json]: We can probably shorten this to something like:
```suggestion
          "description": "Name of SSH private key secret from Buildkite Secrets to use for git clone/fetch operations. The secret must contain a valid SSH private key.",
```

File: schema.json [hunk 2]
```diff
@@ -227,6 +227,12 @@
             { "fetch": "--prune --tags" },
             { "clone": "", "fetch": "" }
           ]
+        },
+        "ssh_secret": {
```
Comment (petetomasik) [inline schema.json:236]: Should we add `maxLength: 255` and perhaps `pattern`/`not.pattern` here to align with [valid characters for Secrets keys](https://buildkite.com/docs/pipelines/security/secrets/buildkite-secrets#create-a-secret-using-the-buildkite-interface)?

File: test/schema.test.js
Comment (petetomasik) [inline test/schema.test.js:1]: Given that the only [valid characters](https://buildkite.com/docs/pipelines/security/secrets/buildkite-secrets#create-a-secret-using-the-buildkite-interface) for Secrets Key are:

> ...which can only contain letters, numbers, and underscores, as valid characters

Many of the values for `ssh_secret` in these tests are (incorrectly) marked as valid.

File: test/valid-pipelines/checkout.yml
Comment (petetomasik) [inline test/valid-pipelines/checkout.yml:1]: `ssh_secret` is missing from pipeline-level `checkout`.

File: test/schema.test.js
Comment (petetomasik) [inline test/schema.test.js:1]: Test is missing that accepts pipeline-level `checkout.ssh_secret`.

File: test/valid-pipelines/checkout.yml [hunk 3]
```diff
@@ -33,15 +33,20 @@ steps:
   - command: test
     checkout:
       skip: false
-      commit_verification: "strict"
+      commit_verification: "strict"
```
Comment (petetomasik) [inline test/valid-pipelines/checkout.yml]: Remove trailing whitespace:

```suggestion
      commit_verification: "strict"
```

File: schema.json [hunk 4]
```diff
@@ -227,6 +227,12 @@
...
+          "type": "string",
+          "description": "Name of SSH private key secret from Buildkite Secrets to use for git clone/fetch operations. The secret must contain a valid SSH private key. The key will be written to a temporary file with 0400 permissions and removed after checkout completes.",
+          "minLength": 1,
+          "examples": ["deploy-key", "github-readonly", "bitbucket-ssh-key"]
```
Comment (petetomasik) [inline schema.json]: Examples are not valid (hyphens->underscores):

```suggestion
          "examples": ["deploy_key", "github_readonly", "bitbucket_ssh_key"]
```

**Review by petetomasik — CHANGES_REQUESTED** [review]
Some suggestions, some issues. Mainly with valid/invalid characters for Secrets Key. The README also requires an update.

File: schema.json (hunk 2 above)
Comment (lizrabuya) [reply schema.json:236]: @petetomasik , just to confirm, all validations mentioned here will apply for ssh_secrets including : 
- If you attempt to use any other characters for the key, or you begin your key with buildkite or bk (regardless of case), your secret will not be created when selecting Create Secret.`

Comment (petetomasik) [reply schema.json:236]: @lizrabuya - yes, we'd like to have these same hints available to users in their IDEs via the schema

File: test/schema.test.js [hunk 5]
```diff
@@ -228,6 +228,189 @@ describe("schema.json", function () {
...
+    const ajv = new Ajv({ allErrors: true });
+    const v = ajv.compile(schema);
+    const pipeline = {
+      steps: [
+        { command: "echo hello", checkout: { ssh_secret: "a".repeat(255) } },
+      ],
+    };
+    expect(v(pipeline)).to.eql(true);
+  });
+
```
Comment (petetomasik) [inline test/schema.test.js:492]: Can we also add a test for `maxLength`? Something like `"a".repeat(256)` (or similar)?

**Comment by lizrabuya** [comment]
@petetomasik , thanks for your patience in reviewing this PR and for highlighting the missing tests. Here's a summary to the recent fixes:
- modified validation for the ssh_secret to be consistent with the secret naming convention _(start with a letter, and only contain letters, numbers, and underscores)_
- updated the tests to reflect the new pattern validation and max length
- updated the tests to include pipeline level addition for checkout.ssh_secret

**Review by petetomasik — APPROVED** [review]
